ddtrace/llmobs/_integrations/crewai.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.llmobs import LLMObs
+from ddtrace.llmobs._constants import INPUT_VALUE
+from ddtrace.llmobs._constants import METADATA
+from ddtrace.llmobs._constants import OUTPUT_VALUE
+from ddtrace.llmobs._constants import ROOT_PARENT_ID
+from ddtrace.llmobs._constants import SPAN_KIND
+from ddtrace.llmobs._constants import SPAN_LINKS
+from ddtrace.llmobs._integrations.base import BaseLLMIntegration
+from ddtrace.llmobs._utils import _get_nearest_llmobs_ancestor
+from ddtrace.trace import Pin
+from ddtrace.trace import Span
+
+
+log = get_logger(__name__)
+
+
+class CrewAIIntegration(BaseLLMIntegration):
+    _integration_name = "crewai"
+    _traces_to_task_span_ids = {}  # maps trace_id to list of task span_ids
+    _traces_to_tasks = {}  # maps trace_id to dictionary of task_id to span_id and span_links
+
+    def trace(self, pin: Pin, operation_id: str, submit_to_llmobs: bool = False, **kwargs: Dict[str, Any]) -> Span:
+        if kwargs.get("_ddtrace_ctx"):
+            tracer_ctx, llmobs_ctx = kwargs["_ddtrace_ctx"]
+            pin.tracer.context_provider.activate(tracer_ctx)
+            if self.llmobs_enabled:
+                LLMObs._instance._llmobs_context_provider.activate(llmobs_ctx)
```

Comment:
Should we use signal handling instead of importing LLMObs here? https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/llmobs/_llmobs.py#L380